### PR TITLE
scheduler(ticdc): fix invlaid checkpoint when redo enabled

### LINKED ISSUE #9774: start table with resolvedTs stored in redo meta when resuming changefeed

### LINKED ISSUE #9830: Table's CheckpointTs can be larger than ResolvedTs
### What did you do?

1. Pause and Resume a changefeed.
2. Kill owner.

### What did you expect to see?

No any error.

### What did you see instead?

4a838da0-3606-4c41-84a8-9952e860fcc6

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
master
```

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #9830, close #9774

### What is changed and how it works?
1. [Init replication set with flushed redo resolvedTs](https://github.com/pingcap/tiflow/pull/9851/files#diff-34b0a3a4c7c7a719f2f0a8e9d837d1666a341db7035322505381c3482a979776R75-R88).
2. [Clenaup resolvedTs and reset it when initlizing changefeed](https://github.com/pingcap/tiflow/pull/9851/files#diff-953746562776cf8aa41cd4a971df8c546649d2721ae787607aed2069f3b44eb0R604).

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -604,6 +601,7 @@ LOOP2:
 		return err
 	}
 	if c.redoMetaMgr.Enabled() {
+		c.resolvedTs = c.redoMetaMgr.GetFlushedMeta().ResolvedTs
```
Comment (nongfushanquan) [inline cdc/owner/changefeed.go:604]: if it is  a new changefeed , what is the value of resolved ts (meta) stored in  s3?

File: cdc/scheduler/internal/v3/scheduler/scheduler_basic.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func (b *basicScheduler) Name() string {
 }
 
 func (b *basicScheduler) Schedule(
-	checkpointTs model.Ts,
+	checkpointTs tablepb.Checkpoint,
```
Comment (nongfushanquan) [inline cdc/scheduler/internal/v3/scheduler/scheduler_basic.go:52]: why not changing the variables name 'checkpointTs" to 'checkpoint' too?

File: cdc/scheduler/internal/v3/scheduler/scheduler_basic.go [hunk 3]
```diff
@@ -140,16 +140,16 @@ func (b *basicScheduler) Schedule(
 // newBurstAddTables add each new table to captures in a round-robin way.
 func newBurstAddTables(
 	changefeedID model.ChangeFeedID,
-	checkpointTs model.Ts, newSpans []tablepb.Span, captureIDs []model.CaptureID,
+	checkpointTs tablepb.Checkpoint, newSpans []tablepb.Span, captureIDs []model.CaptureID,
```
Comment (nongfushanquan) [inline cdc/scheduler/internal/v3/scheduler/scheduler_basic.go:143]: The same as above

File: cdc/owner/changefeed.go (hunk 1 above)
Comment (CharlesCheung96) [reply cdc/owner/changefeed.go:604]: The meta file is created when the redoMetaManager is initialized,  ref: https://github.com/pingcap/tiflow/blob/master/cdc/redo/meta_manager.go#L206-L208.
So, if it is a new changefeed, the value of meta will be equal to startTs

**Review by nongfushanquan — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Pull request merged**